gradle-conjure/src/main/java/com/palantir/gradle/conjure/ConjurePlugin.java

```diff
@@ -146,11 +146,8 @@ private static void setupConjureJavaProject(
             ConjureProductDependenciesExtension productDependencyExt) {
         if (JAVA_PROJECT_SUFFIXES.stream()
                 .anyMatch(suffix -> project.findProject(project.getName() + suffix) != null)) {
-            Configuration conjureJavaConfig = project.getConfigurations().maybeCreate(CONJURE_JAVA);
-            File conjureJavaDir = new File(project.getBuildDir(), CONJURE_JAVA);
-            project.getDependencies().add(CONJURE_JAVA, CONJURE_JAVA_BINARY);
-            ExtractExecutableTask extractJavaTask = ExtractExecutableTask.createExtractTask(
-                    project, "extractConjureJava", conjureJavaConfig, conjureJavaDir, "conjure-java");
+            ExtractExecutableTask extractJavaTask = ExtractExecutableTask.getOrCreateExtractTaskOnRootProject(
```

Comment:
I would create a separate plugin that registers the task and have this snippet can apply the plugin on the root project

Reply:
Makes sense, does gradle deduplicate plugin registration if it's called several times?

Reply:
yes it handles it for you!